plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/handler/ServerAdapterHandler.java

```diff
@@ -53,7 +54,27 @@
 	public Object execute(final ExecutionEvent event) throws ExecutionException {
 		final IWorkbenchWindow workbenchWindow = HandlerUtil.getActiveWorkbenchWindow(event);
 		final ISelection selection = HandlerUtil.getCurrentSelection(event);
-		final IResourceUIModel selectedResource = UIUtils.getFirstElement(selection, IResourceUIModel.class);
+		IResourceUIModel selectedResource = UIUtils.getFirstElement(selection, IResourceUIModel.class);
+		if(selectedResource == null && (selection instanceof IStructuredSelection)) {
+			//Object may be selected in Properties view, with single selection.
+			final Object selected = ((IStructuredSelection)selection).getFirstElement();
+			if(selected instanceof IService || selected instanceof IRoute) {
+				//Method getOpenShiftServer(IResourceUIModel) will do well with just a simple adapter.
+				selectedResource = new IResourceUIModel() {
+
+					@Override
+					public Object getParent() {
+						return null;
+					}
+
+					@Override
+					public IResource getResource() {
+						return (IResource)selected;
+					}
+
+				};
+			}
+		}
 		final IServer openShiftServer = getOpenShiftServer(selectedResource);
```

Comment:
I'd rather prefer you changed the getOpenShiftServer signature to take a IResource parameter instead of a IResourceUIModel